### PR TITLE
Lower level api v2

### PR DESCRIPTION
**Pull request by brumar**
I preferred a second version to avoid messing with the discussion on the first one.
Improvements: 
- optional support for async if the tts backend supports it (I think this is were async has the most benefits), fallback to threads if async not supported and n_jobs > 1
- simplify the podcast_stage_decorator
- finally requires the llm_backend to send tuples so that parsing the string is not the responsibility of the podcast class. That would be less ackward when structured output will be used.
- documentation
- from_transcript method to provide an escape hatch if transcript have been saved or obtained with another method

### CODE REVIEW

**Comment by brumar** [comment]
@souzatharsis the multiple feature requests like "integrate with" make it somehow urgent to abstract away llm and tts backend with well thought ABCs. This way, it will be easy to publish a guide on "how to integrate with your own <<thing>>" and probably setup a contribs folder or even repository later on. 

Maybe, to get closer to the domain model, I am starting to think that the concept of Character is central to this podcast thing. Once defined, they could be the second argument for each backend (tts and llm). This way, this design won't feel too generic and closer to the domain model.

File: podcastfy/core/podcast.py [hunk 1]
```diff
@@ -0,0 +1,410 @@
...
+            content (str): The raw content to be processed into a podcast.
+            llm_backend (LLMBackend): The language model backend for generating the transcript.
+            tts_backend (TTSBackend): The text-to-speech backend for converting text to audio.
+            default_tts_n_jobs (int, optional): The default number of concurrent jobs for TTS processing.
+                Defaults to 1.
+
+        Raises:
+            logging.warning: If using a synchronous TTS backend with multiple jobs.
+        """
+        self.content = content
```
Comment (souzatharsis) [inline podcastfy/core/podcast.py:159]: podcastfy v0.2.1+ is now multimodal, meaning content should be a list of elements, e.g. text or image for now but video and audio too in the future.

File: podcastfy/core/podcast.py [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+    class DummyTTSBackend(TTSBackend):
+        def text_to_speech(self, text: str, voice: str) -> Path:
+            with NamedTemporaryFile(suffix=".mp3", delete=False) as temp_file:
+                PydubAudioSegment.silent(duration=1000).export(temp_file.name, format="mp3")
+            return Path(temp_file.name)
+
+
+    # Initialize the podcast
+    podcast = Podcast(
+        content="""
```
Comment (souzatharsis) [inline podcastfy/core/podcast.py:346]: should be multimodal

File: podcastfy/core/podcast.py [hunk 3]
```diff
@@ -0,0 +1,410 @@
...
+    podcast.build_audio_segments()
+    print(f"After building audio segments: {podcast.state}")
+    print(f"Number of audio segments: {len(podcast.audio_segments)}")
+
+    # Step 3: Stitch audio segments
+    podcast.stitch_audio_segments()
+    print(f"After stitching audio: {podcast.state}")
+
+    # Rework example: modify the transcript and rebuild (auto_finalize is True by default)
+    with podcast.rework(PodcastState.TRANSCRIPT_BUILT):
```
Comment (souzatharsis) [inline podcastfy/core/podcast.py:370]: .rework is an exceptional idea!

**Review by souzatharsis — CHANGES_REQUESTED** [review]
This is a much need update! Added comments pertaining to multimodality. Otherwise, we should push a v1 of this ASAP.

File: podcastfy/core/podcast.py (hunk 1 above)
Comment (brumar) [reply podcastfy/core/podcast.py:159]: Indeed!

**Comment by brumar** [comment]
@souzatharsis Great! I think that when this is really integrated in the code, other design consideration would takes place. Did you see my comment about the Character thing? If you think that it's worth a shot, I integrate this aspect into my PR.

**Comment by souzatharsis** [comment]
Yes, please integrate Character. very interesting idea.

**Comment by souzatharsis** [comment]
Should we close this one and consolidate it under #61 ?

**Pull request closed without merging**